### PR TITLE
Feat: prevent page scrolling on mouse wheel

### LINKED ISSUE #8: Mouse wheel and keyboard also does scroll the page vertically
First: Great component, great styling options, and many thanks for sharing.

Problem:
However, the mouse wheel and keyboard support is not perfect. Although it works ( can change the value with both the mouse wheel, when hovering, and the up/down arrow keys when focused), the page ALSO gets vertically scrolled at the same time.
This does not happen with the "precise" mode, using the shift keys, probably because these are not handled at the page level anyway.

Possible solution:
There should be, by default, no further propagation of these events, so that scrolling does not occur by default. 

This should be configurable however, since it is conceivable that someone wants to handle these events outside.

### PR DESCRIPTION
**Pull request by suterma**
Finally worked through it. 

Please note: I tried to add some cypress tests (a first for me, and I kind of liked it), but I did not manage to have "real" keypresses or mouse wheel events, that actually break things as expected. So, in the current form, they are quite useless. You might want to fix them, if possible, or just remove them.

Best Regards, Marcel

Fixes #8

### CODE REVIEW

**Comment by Uninen** [comment]
Thank You for your work, this looks much better 👍 

Don't worry about the tests. I didn't have time to write proper ones for the current implementation so it's hard to add more on top. In addition, I'm planning to convert all browser-tests to using Playwright ([for resons](https://til.unessa.net/weeknotes/2022-18/)) so I need to rework everything in any case.

File: src/ControlKnob.vue [hunk 1]
```diff
@@ -130,6 +132,7 @@ const downListener = (event: MouseEvent) => {
   mouseIsDown.value = true
   mouseMoved.value = false
   prevY = event.clientY
+  preventScrolling(event);
```
Comment (Uninen) [inline src/ControlKnob.vue:135]: Remove semicolons.

Feel free to add `.prettierrc` file to the root of the repo if it helps to guide your editor better. Here's the content for that file: 

```
{
  "trailingComma": "es5",
  "tabWidth": 2,
  "semi": false,
  "singleQuote": true,
  "printWidth": 100
}
```

File: src/ControlKnob.vue [hunk 2]
```diff
@@ -168,6 +171,16 @@ function moveListener(event: MouseEvent) {
 
 const debouncedMoveListener = leadingDebounce(moveListener)
 
+/** According to the chosen option, prevents propagation of the event.
+ * @remarks If set, keeps page from scrolling while handling the knob
+ */
+function preventScrolling(event: TouchEvent | MouseEvent | KeyboardEvent): void {
+  if (passiveEvents === false) {    
+    event.preventDefault();
```
Comment (Uninen) [inline src/ControlKnob.vue:179]: Remove semicolons.

File: src/ControlKnob.vue [hunk 3]
```diff
@@ -202,10 +215,9 @@ function keyDownListener(event: KeyboardEvent) {
 
   if (
     hasFocus.value &&
-    shiftModifier.value &&
     (event.key === 'ArrowUp' || event.key === 'ArrowDown')
   ) {
-    event.preventDefault()
+    preventScrolling(event);
```
Comment (Uninen) [inline src/ControlKnob.vue:220]: Remove semicolons.

File: src/ControlKnob.vue [hunk 4]
```diff
@@ -254,7 +267,8 @@ watch(
   (element, oldElement) => {
     if (element && !oldElement) {
       element.addEventListener('mousedown', downListener)
-      element.addEventListener('wheel', wheelListener)
+      // Note: The wheel event is handeled according to the option, possibly manipulating scrolling behaviour
```
Comment (Uninen) [inline src/ControlKnob.vue:270]: While this is a good and descriptive comment, I think it's obvious for everyone reading the code that the event is handled by the options. Remove this comment.

File: tests/e2e/integration/scroll_spec.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+describe('Homepage', () => {
```
Comment (Uninen) [inline tests/e2e/integration/scroll_spec.ts:1]: Thanks for the effort on this one 👍

**Review by Uninen — CHANGES_REQUESTED** [review]
I'm happy to merge after these small changes 👍

**Comment by suterma** [comment]
Incorporated the requested changes. Thanks for the prettier formatting rules, this now works much better for me.

**Review by Uninen — APPROVED** [review]
Looks good. Thank You again for the contribution 👍

**Pull request merged**